### ISSUE TITLE
Update process is bad UX

### ISSUE THREAD

**Issue by joeytwiddle** [macvim-dev/macvim#390]
Surely I'm not the only one who felt this?  I am using the binary download of MacVim (because that gave me better Python support than the homebrew install did).

The real problem with the process is that it asks me when I am in the middle of using Vim combined with the fact that after the download is complete the prompt then wants me to restart Vim immediately, and cannot be closed.

Some possible improvements would be:

- Download (or offer to download) the update in the background, and cache it without applying it.
- Do not force me to restart immediately after the download is complete.
- Ask me whether to update when I first open Vim, before I have invested any time and state in opening and editing files.
- Or personally my preference would be: Silently download in the background, and automatically update the next time I restart Vim, without any prompting at all.

I think one or more of these could smooth off the sharp edges. Everything else about MacVim has been great for me!

**Comment by splhack** [comment]
It's the behavior of Sparkle https://sparkle-project.org/ and you can select "Remind Me Later". Also you can disable auto update https://github.com/macvim-dev/macvim/wiki/FAQ#how-can-i-disable-autoupdate-sparkle .

**Comment by joeytwiddle** [comment]
Thanks, good to know.

They [mention](https://github.com/sparkle-project/Sparkle) "automatic background updates" and "user can choose to automatically download and install all updates" but I don't see how to enable those feature for MacVim.

However I did find in MacVim's preferences that "Check for updates" (when MacVim launches) was disabled, so I have turned that on. It will probably be a small improvement on what I experienced before, but I would still love to download in the background, and automatically update when Vim opens (or closes).

**Comment by joeytwiddle** [comment]
Thanks, good to know.

They [mention](https://github.com/sparkle-project/Sparkle) "automatic background updates" and "user can choose to automatically download and install all updates" but I don't see how to enable those feature for MacVim.

However I did find in MacVim's preferences that "Check for updates" (when MacVim launches) was disabled, so I have turned that on. It will probably be an improvement on what I experienced before, but I would still love to download in the background, and automatically update when Vim opens (or closes).
